### DISCUSSION TITLE
Wix 3.11 bootstrapper uninstallation of all of the ExePackages inside

### DISCUSSION THREAD

**Discussion by JETMikeMc** [wixtoolset/issues#8861] [Questions]
Hello, I have a master installer bundle that calls other WiX installers. However on the uninstallation, I would like the uninstall operation to uninstall all of the child packages.

I am using a bootstrapper with a bundle.  I am using a format like:

<ExePackage
	SourceFile="..\dir\dir\\bin\Release\MyMasterInstaller.exe"
		InstallCommand="/quiet" UninstallCommand="/uninstall /quiet" >
</ExePackage>

All of the other installers have this convention set up as well. However on the uninstall process, only the top level application is removed and not the other ones.  How do  I go about uninstalling items that are chained?

**Comment by robmen** [comment]
Use the `ExePackage` elements `DetectCondition` attribute. It was a bug that attribute was not required in WiX v3.

In WiX v4+, you can use `BundlePackage` instead of `ExePackage`, and WiX will automatically set the `DetectCondition` for you because it understands bundles.

**Comment by JETMikeMc** [comment]
What is the syntax for DetectCondition?  Does this work in conjunction with UninstallCommand ?


[cid:a1c0f739-53c9-47a6-88da-f24db4788f2c]

________________________________
